test/e2e/handler/utils.go

```diff
@@ -61,22 +61,26 @@ func interfaceByName(interfaces []interface{}, searchedName string) map[string]i
 	return dummy
 }
 
-func setDesiredStateWithPolicyAndNodeSelector(name string, desiredState nmstate.State, nodeSelector map[string]string) {
+func updateNodeNetworkConfigurationPolicy(name string, desiredState nmstate.State, nodeSelector map[string]string) error {
 	policy := nmstatev1beta1.NodeNetworkConfigurationPolicy{}
 	policy.Name = name
 	key := types.NamespacedName{Name: name}
-	Eventually(func() error {
-		err := testenv.Client.Get(context.TODO(), key, &policy)
-		policy.Spec.DesiredState = desiredState
-		policy.Spec.NodeSelector = nodeSelector
-		policy.Spec.Parallel = parallelRollout
-		if err != nil {
-			if apierrors.IsNotFound(err) {
-				return testenv.Client.Create(context.TODO(), &policy)
-			}
-			return err
+	err := testenv.Client.Get(context.TODO(), key, &policy)
+	policy.Spec.DesiredState = desiredState
+	policy.Spec.NodeSelector = nodeSelector
+	policy.Spec.Parallel = parallelRollout
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return testenv.Client.Create(context.TODO(), &policy)
 		}
-		return testenv.Client.Update(context.TODO(), &policy)
+		return err
+	}
+	return testenv.Client.Update(context.TODO(), &policy)
+}
+
+func setDesiredStateWithPolicyAndNodeSelector(name string, desiredState nmstate.State, nodeSelector map[string]string) {
```

Comment:
Let's rename this as `setDesiredStateWithPolicyAndNodeSelectorEventually` is not a big refactor I think there are just three places.

Reply:
done